Felienne/hedy
# Ace can highlight keywords in other languages (currently dutch and english)

**Description**

- Adds a  syntaxModesRules-'lang'.ts file for English and Dutch.
- Each Rules-'lang' file generates its own mode file as ace/mode/level1en.js for example as opposed to just ace/mode/level1.js
- slight changes in app.ts and index.ts to allow for these different highlighting rules to actually be used (importing the syntaxModesRules-'lang'.ts files and calling the correct ace/mode file), also includes an enable to turn this on or off. It is currently off by default.

**Fix for**

Closes #909 

**How to test**

Set local_keywords_highlights_enabled in app.ts to true and rebuild the appbundle, afterwards run the website. You should now see that the keywords highlighted for the current language (currently just dutch and english) are highlighted appropriately according to the language's keywords as given in 'lang'.yaml

**Some Notes**

-Visual Studio Code (And likely other developement environments) will complain about duplicate function implementations within the various syntaxModesRules files, since these were generated with the same methods. This doesn't seem to cause any problems at runtime however.
-I also made a script to easily generate a syntaxRulesModes-'lang' file from the keywords saved in the .yaml files. I didn't include it here because it would only bloat the pull request out and wasn't strictly necessary, but if there's an interest I can share it.
